lib/Component.spec/validateProps.spec.lua

```diff
@@ -0,0 +1,172 @@
+return function()
+    local createElement = require(script.Parent.Parent.createElement)
+    local createReconciler = require(script.Parent.Parent.createReconciler)
+    local createSpy = require(script.Parent.Parent.createSpy)
+    local NoopRenderer = require(script.Parent.Parent.NoopRenderer)
+
+    local Component = require(script.Parent.Parent.Component)
+
+    local noopReconciler = createReconciler(NoopRenderer)
+
+    it("should be invoked when mounted", function()
+        local MyComponent = Component:extend("MyComponent")
+
+        local validatePropsSpy = createSpy(function()
+            return true
+        end)
+
+        MyComponent.validateProps = validatePropsSpy.value
+
+        function MyComponent:render()
+            return nil
+        end
+
+        local element = createElement(MyComponent)
+		local hostParent = nil
```

Comment:
This file has some weird mixed whitespace shenanigans going on. Let's clean that up!